.gitlab-ci.yml

```
@@ -13,14 +13,22 @@ stages:
 
 build-runner-image:
   stage: build
-  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/docker:20.10.13
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci-containers-project:v2.0.0
   tags: ["arch:amd64"]
   variables:
     RELEASE_IMAGE: "false"
+    GITHUB_PRIVATE_KEY_PATH: /tmp/github.pem
   rules:
     - when: on_success
+  before_script:
+    - export GITHUB_APP_ID=$(aws ssm get-parameter --region us-east-1 --name ci.test-infra-definitions.github_app_id --with-decryption --query "Parameter.Value" --out text)
+    - export GITHUB_INSTALLATION_ID=$(aws ssm get-parameter --region us-east-1 --name ci.test-infra-definitions.github_installation_id --with-decryption --query "Parameter.Value" --out text)
+    - aws ssm get-parameter --region us-east-1 --name ci.test-infra-definitions.github_private_key --with-decryption --query "Parameter.Value" --out text > $GITHUB_PRIVATE_KEY_PATH
+    - export JWT=$(.github/jwt.sh $GITHUB_APP_ID $GITHUB_PRIVATE_KEY_PATH)
```

This will leak a file with a private key in a world readable directory.

The best would be to keep it only in memory, in an environment variable.

The `<(…)` bash construct can be used to send the output of a command to a program that expects a filename.

```suggestion
    - export GITHUB_PRIVATE_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.test-infra-definitions.github_private_key --with-decryption --query "Parameter.Value" --out text)
    - export JWT=$(.github/jwt.sh $GITHUB_APP_ID <(echo $GITHUB_PRIVATE_KEY))
```